pithos/pithos.py

```diff
@@ -366,6 +368,23 @@ def get_proxy(self):
 
         return None
 
+    def on_explicit_content_filter_checkbox(self, *ignore):
+        if self.pandora.connected:
+            def fn(*ignore):
+                if self.filter_state != self.prefs_dlg.explicit_content_filter_checkbutton.get_active():
+                    self.pandora.set_explicit_content_filter_to(self.prefs_dlg.explicit_content_filter_checkbutton.get_active())
+
+            def callback(*ignore):
+                if self.prefs_dlg.explicit_content_filter_checkbutton.get_active():
+                    logging.info('Getting new playlist.')
+                    self.waiting_for_playlist = False
+                    self.stop()
+                    self.current_song_index = None
+                    self.songs_model.clear()
+                    self.get_playlist(start = True)
+
+            self.worker_run(fn, '', callback)
```

Comment:
self.prefs_dlg.explicit_content_filter_checkbutton.get_active() just polls the state of the check box. It doesn't actually change anything in the UI.

Basically what's going on is we're comparing the filter state we got when we opened the prefs to the state of checkbox when apply is clicked. If it different we set the filter to the checkbox state. If it's not different we do nothing.